### PR TITLE
First version of new Global Mission Partners: India pages

### PR DESCRIPTION
**Pull request by gerv**
plus other tweaks to reflect the way MOSS is now running

## Description
MOSS is launching a new track focussed on India; this content supports that launch. It needs to be reviewed and staged, but not pushed live until the launch date, which will be in early August. (We hope a Hindi translation of some of this content will follow.)

## Issue / Bugzilla link
None

## Testing
Tested locally by me

### CODE REVIEW

**Comment by gerv** [comment]
@pmac or @craigcook: what's the ETA here? I'd like to get this page translated, that normally has a 2-week lead time, and we want to launch in early August. So we are bumping up against deadlines...

Thanks for any impetus you can give this :-)

**Comment by ejregithub** [comment]
@gerv - If the PR is in (it clearly is) and the copy is deemed final, l10n can begin when those 2 conditions are met. Can you please confirm copy state? We can then reach out to @peiying2 w. request for localization, indicating locale preferences.

**Comment by gerv** [comment]
I thought the tools they use for localising www.mozilla.org can only work with committed text?

I have notified the Hindi translation team directly of my need for this to be translated, but I wasn't able to tell them how, technically, to go about translating it.

**Comment by peiying2** [comment]
@gerv we have never localized any MOSS pages at this point.  Two questions:
- MOSS India-partner, we can make it available in Hindi only if this is desired.  
- MOSS landing page: would you like this to be in Hindi only? Or should we make it available for other communities.  

"can only work with committed text?" - We strip the text from the codes, put it in l10n tools, so the communities only see the strings and some notes about context. We need time to prep the file.  

There are lots of updates and new pages on mozilla.org at the moment, in the middle of summer when people are taking time off, so the communities are pretty busy with that.  If MOSS landing is opened up to the communities, we can make it an optional page, which means if they have bandwidth, after finishing other competing projects, they can choose the optional pages to work on.

**Comment by peiying2** [comment]
@gerv who from the India community did you talk with?  The l10n community still has tons of work to complete for what they ought complete: https://l10n.mozilla-community.org/webdashboard/?locale=hi-IN

**Comment by gerv** [comment]
@peiying2: my contact is Umesh Agarwal, who, according to https://wiki.mozilla.org/L10n:Teams:hi-IN, is the overall owner of hi-IN l10n. It may well be that they also have other work to do, but this is an important Mozilla initiative in India, and deserves high prioritization. That team has already localised our application form, wiki pages and press release; the web pages are the last (and one of the most important) components.

I am looking for both the main MOSS page, and the Mission Partners: India page to be available in both English and Hindi. Testing out our processes for multiple languages is one of the purposes of this pilot program. If this is too much work, the Mission Partners: India page is higher priority.

Please can you do whatever is necessary to prepare these page(s) for localization, or tell me how to do it? It seems unfortunate to me that each page requires special work; I would expect l10n to be a first-class citizen for the main Mozilla website, and something that is easy for people to do... :-|

**Comment by peiying2** [comment]
@gerv Our team works with Umesh closely on a variety of projects, so it is good to hear you two have touched base on this new project.  Once the pages are ready, we will let him and team know.  

We can enable the pages for Hindi for the two pages.  We will follow the standard process we have to prep all mozilla.org pages, nothing less - the process that localizers are familiar with, and they are easy for the contributors to localize.    Make sure English is targeted to international audience.  This usually is the most challenging part for the localizers.

**Comment by craigcook** [comment]
Sorry I've neglected this PR. I can review it today.

**Comment by gerv** [comment]
@peiying2: great :-) Let me know if you need anything more from me. Our launch date is 31st July...

**Comment by peiying2** [comment]
@gerv Is the launch in English on July 31st?  Then Hindi in early August - what is the hard date?

**Comment by gerv** [comment]
Sorry for being unclear. We are launching on July 31st, and all versions need to be ready by then. Obviously, we want to launch in both languages at once :-)

**Comment by flodolo** [comment]
Can't spot any particular issue.

Can you add this line in base.html? https://github.com/gerv/bedrock/blob/4bc9ce44c5fff13d2ab87fb851cbf781e6c52b37/bedrock/mozorg/templates/mozorg/moss/base.html#L6

```
{% add_lang_files "mozorg/moss/index" %}
```

This allows us to use that file to load strings

**Comment by peiying2** [comment]
@gerv the files are enabled for localization. Umesh is notified that he and community can start working on them.

**Comment by gerv** [comment]
@craigcook: I'm supposed to be on holiday this week; could you possibly sort this out for me? Thanks :-)

**Comment by peiying2** [comment]
@gerv could you provide the staging link for the partner page?  We need it for testing.

**Comment by flodolo** [comment]
There is no staging link until this is merged…

At this point I'm going to fix it myself, and merge if it passes the tests, then it will be available on dev.

**Comment by craigcook** [comment]
Thanks @flodolo!

**Comment by peiying2** [comment]
@gerv Both files are translated in Hindi.  Could you state the staging info for the partner page so the community can have a chance to review it?

**Comment by gerv** [comment]
peiying: the launch has been delayed due to a legal issue; I'll circle back next week.

But I don't understand your question; what do you mean by "state the staging info"? I assume the page would be put up on whatever staging server is used for mozilla.org, but that's not something I can do...

**Comment by flodolo** [comment]
> peiying: the launch has been delayed due to a legal issue; I'll circle back next week.

That page is currently enabled in prod, is that not part of the plan?
https://www.mozilla.org/en-US/moss/mission-partners-india/

**Comment by gerv** [comment]
No, not part of the plan. As the initial description here says: " It needs to be reviewed and staged, but not pushed live until the launch date, which will be in early August." Can this be reverted ASAP, please? CC @flodolo @craigcook @pmac.

**Comment by flodolo** [comment]
I merged based on https://github.com/mozilla/bedrock/pull/4973#pullrequestreview-51563255

I don't think this can be put behind a flag at this point. I suggest to open a bug for visibility.

**Comment by gerv** [comment]
If it can't be put behind a flag, we need to back it out. This initiative is not launching for another month.

I am very surprised that mozilla.org doesn't have the easy ability to launch stuff at a particular time, and I'm also surprised that a clear statement in an issue about launch dates and about not putting it live doesn't have the required effect in terms of process. What should I have done differently?

**Comment by alexgibson** [comment]
I haven't looking into this pull request until now but I can say the following:

- Mozorg does have a system to toggle content on/off via switches. If you want to use it, we can arrange it. Hopefully we can fix this up without having to revert the commit.
- This change does not seem to have any sort of bug associated with it, which is probably should.
- Reading the discussion here, this PR also seemed to come out of the blue to our l10n team. It probably should have been communicated clearer and ahead of time.
- If this PR should not have been merged, it should have had a clear do-not-merge label on it and also a clear due date.
- If this was required to put put up on a demo server before merging, we could have done this for you (staging is as close to prod as possible, it's not a dev environment).

TL:DR Merging this early was obviously a mistake, but it could have probably been avoided through better communication on both sides.

**Comment by gerv** [comment]
> This change does not seem to have any sort of bug associated with it, which it probably should.

Where is the process for making updates to mozilla.org documented, that tells me that I should have filed a bug? And could that documentation be linked from the build instructions?

> Reading the discussion here, this PR also seemed to come out of the blue to our l10n team. It probably should have been communicated clearer and ahead of time.

I have been in contact with the hi-IN l10n team on this topic for at least a month; they knew this was coming.

> If this PR should not have been merged, it should have had a clear do-not-merge label on it and also a clear due date.

Again, point me at the documentation which I should have read which told me I should have done that.

> If this was required to be put up on a demo server before merging, we could have done this for you (staging is as close to prod as possible, it's not a dev environment).

Well, was a demo server required? I didn't require one. There seems to be some suggestion that one is required for l10n, but that seems unlikely to me - I doubt you set up a demo server for each checkin where people plan to localise it...

**Comment by alexgibson** [comment]
> Where is the process for making updates to mozilla.org documented, that tells me that I should have filed a bug? And could that documentation be linked from the build instructions?

The docs section on [contributing to bedrock]( http://bedrock.readthedocs.io/en/latest/contribute.html#git-commit-messages) specify that all commits should be referenced with a Bugzilla bug number. The pull request template also has a section ready for you to drop a bug number into. While we don't always insist on a bug for very minor changes, the amount of discussion here says to me this probably warranted someone creating a bug. We could probably make this clearer, but it should at least raise a question that perhaps you should file a bug.

> I have been in contact with the hi-IN l10n team on this topic for at least a month; they knew this was coming.

That's not our websites l10n team, who are the team responsible for co-coordinating translation on bedrock.

> Again, point me at the documentation which I should have read which told me I should have done that.

I said communication should have been better on both sides, this should have been done by someone on our team.

> Well, was a demo server required? I didn't require one. There seems to be some suggestion that one is required for l10n, but that seems unlikely to me - I doubt you set up a demo server for each checkin where people plan to localise it...

If you didn't want this merged but wanted l10n to start, yes this is what should have happened. If you talked to our l10n websites team about this ahead of time, then the wants/needs probably would have been cleared here.

**Comment by flodolo** [comment]
> I have been in contact with the hi-IN l10n team on this topic for at least a month; they knew this was coming.

The hi-IN team is in charge of localizing content in Hindi once exposed for localization. The latter is managed by l10n-drivers: https://wiki.mozilla.org/L10n:Mozilla_Team

> Well, was a demo server required? I didn't require one. There seems to be some suggestion that one is required for l10n, but that seems unlikely to me - I doubt you set up a demo server for each checkin where people plan to localise it...

How else are localizers supposed to test their translations? Either it's merged to master, or pushed to a demo server.

P.S. I think we just got rid of the issue template, which might not have helped here.

**Comment by gerv** [comment]
> The latter is managed by l10n-drivers: https://wiki.mozilla.org/L10n:Mozilla_Team

You mean there's some special process before each new piece of content can be localized? I struggle to think of why that would be necessary - can you enlighten me? We don't have a special process before each new string in Firefox can be localized.

> How else are localizers supposed to test their translations? Either it's merged to master, or pushed to a demo server.

Well, in order to make this patch, I had to download, install and get running the bedrock codebase.

If you say "well, that's far too complicated", I'd say "Precisely! So why do we have a process for updating content on mozilla.org which involves lots of people having to do it?"

How much of www.mozilla.org is actually translated, and how current are those translations? It's our flagship website, so I'd hope those numbers would be high. But if it's this complicated, perhaps they aren't.

**Comment by alexgibson** [comment]
This discussion is no longer being productive. Please file a bug to sort out this issue.

**Comment by flodolo** [comment]
> You mean there's some special process before each new piece of content can be localized? I struggle to think of why that would be necessary - can you enlighten me? We don't have a special process before each new string in Firefox can be localized.

mozilla.org: strings need to be extracted from Django templates, checked for errors and missing comments, exposed in repositories in a format understood by localization tools, tracked in dashboards. 

mozilla.org might be our flagship website, but it's just a speckle in the amount of content we ask our localizers to work on.

You have an oversimplified look of how localization works at Mozilla, Firefox included, but I don't think this PR is the right place to talk about it, as already stated by Alex ;-)
https://mozilla-l10n.github.io/documentation/

**Comment by gerv** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1385566 filed.
